File: php/shapes/Graph.php
<?php

require_once 'Shape.php';

class Graph
{
    private array $shapes;

    public function __construct(array $shapes)
    {
        $this->shapes = $shapes;
    }

    public function isInGraph(float $x, float $y): bool
    {
        return sizeof(
                array_filter($this->shapes,
                    function (Shape $shape) use ($y, $x): bool {
                        return $shape->isInShape($x, $y);
                    })
            ) > 0;
    }
}

File: src/main/java/com/moleus/web/service/areaCheck/shapes/Graph.java
package com.moleus.web.service.areaCheck.shapes;

import com.moleus.web.service.areaCheck.quadrant.Point;
import lombok.extern.log4j.Log4j2;

import java.util.ArrayList;
import java.util.List;

@Log4j2
public class Graph {
    private final List<Shape> shapes = new ArrayList<>();

    public void addShape(Shape shape) {
        shapes.add(shape);
    }

    public boolean isInGraph(Point point) {
        log.info("Shapes size: {}", shapes.size());
        return shapes.stream().anyMatch(shape -> shape.isInShape(point));
    }
}
